api/s2n.h

```diff
@@ -433,6 +433,9 @@ extern int s2n_async_pkey_op_apply(struct s2n_async_pkey_op *op, struct s2n_conn
 S2N_API
 extern int s2n_async_pkey_op_free(struct s2n_async_pkey_op *op);
 
+S2N_API
+extern int s2n_config_set_cert_req_legacy_dss_enabled(struct s2n_config *config, uint8_t enabled);
```

Comment:
Couple of things:

* Can we make this a `bool` instead?
* Can you add a comment here about it being not a good idea to call this unless you know you really need it?
* I worry about the name indicating something other than it does. It makes me think we're actually enabling DSS, rather than just for GREASEing/compatibility purposes.

Reply:
> Is there any other suggested naming?

I think I would prefer something like `s2n_config_enable_cert_req_dss_legacy_compat`.

> Do you mean returning a bool type?

No i meant the `enabled` argument. Having it as a `uint8_t` doesn't communicate the acceptable value as well as a `bool`. Another option would be to just remove the argument entirely:

```
S2N_API
extern int s2n_config_enable_legacy_cert_req_dss_grease(struct s2n_config *config);
```

This is the direction that the QUIC enabled flag went: https://github.com/awslabs/s2n/blob/5dd7178529c8fd9b72ee39a32dbcb7bfd9d16173/tls/s2n_quic_support.c#L38-L43